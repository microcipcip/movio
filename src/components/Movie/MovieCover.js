import React from 'react'
import PropTypes from 'prop-types' // eslint-disable-line no-unused-vars

const MovieCover = ({ n }) => {
  return <div>MovieCover {n}</div>
}

MovieCover.propTypes = {}
MovieCover.defaultProps = {}

export default MovieCover
