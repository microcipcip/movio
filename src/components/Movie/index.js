import Index from './Movie'

export default Index
