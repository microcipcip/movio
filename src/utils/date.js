export const addZero = date => ('0' + date).slice(-2)

export const formatMovieDate = movieDate => {
  const d = new Date(movieDate)
  return `${addZero(d.getDate())} ${addZero(
    d.getMonth() + 1
  )} ${d.getFullYear()}`
}
