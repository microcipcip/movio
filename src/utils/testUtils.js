import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'styled-components'
import 'jest-styled-components'
import 'jest-dom/extend-expect'
import { genreList, movieList } from 'utils/testData'
import * as s from 'styles/vars'

import CinemaContext, { cinemaContextState } from 'components/App/CinemaContext'

const customRender = (node, data = {}) => {
  return render(
    <CinemaContext.Provider value={Object.assign(cinemaContextState, data)}>
      <ThemeProvider theme={s.theme.dark}>{node}</ThemeProvider>
    </CinemaContext.Provider>
  )
}

const domToString = el => {
  const clnEl = el.cloneNode(true)
  const tmp = document.createElement('div')
  tmp.appendChild(clnEl)
  return tmp.innerHTML
}

// re-export everything
export * from 'react-testing-library'
export { customRender, domToString, genreList, movieList }
