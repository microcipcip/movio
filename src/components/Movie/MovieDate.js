import React from 'react'
import PropTypes from 'prop-types' // eslint-disable-line no-unused-vars
import styled from 'styled-components'
import { formatMovieDate } from 'utils/date'
import { MovieConsumer } from './Movie'

const MovieDate = props => (
  <MovieConsumer>
    {({ release_date }) => (
      <MovieDateStyled datetime={release_date}>
        {formatMovieDate(release_date)}
      </MovieDateStyled>
    )}
  </MovieConsumer>
)

const MovieDateStyled = styled.time`
  color: ${prop => prop.theme.textColorLight};
  font-size: 1.4rem;
`

MovieDate.propTypes = {}
MovieDate.defaultProps = {}

export default MovieDate
