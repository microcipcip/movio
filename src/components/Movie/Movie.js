import React, { createContext } from 'react'
import PropTypes from 'prop-types' // eslint-disable-line no-unused-vars
import checkCompoundConsumer from 'helpers/checkCompoundConsumer'
import MovieCover from './MovieCover'
import MovieTitle from './MovieTitle'
import MovieDate from './MovieDate'

const MovieContext = createContext()

const Movie = ({ children, ...restOfProps }) => (
  <MovieContext.Provider value={restOfProps}>{children}</MovieContext.Provider>
)

Movie.Cover = MovieCover
Movie.Title = MovieTitle
Movie.Date = MovieDate

Movie.propTypes = {}
Movie.defaultProps = {}

// prevent Consumer from being used outside tha parent
export const MovieConsumer = checkCompoundConsumer(
  MovieContext,
  'Compound components of Movie should render beneath it'
)

export default Movie
