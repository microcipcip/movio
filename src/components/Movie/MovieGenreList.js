import React from 'react'
import styled from 'styled-components'
import { MovieConsumer } from './Movie'

const MovieGenreList = props => (
  <MovieConsumer>
    {({ genre_ids, genreList }) => (
      <MovieGenreListStyled>
        {genre_ids.map(genreId => (
          <MovieGenreTag key={genreId}>{genreList[genreId].name}</MovieGenreTag>
        ))}
      </MovieGenreListStyled>
    )}
  </MovieConsumer>
)

const MovieGenreListStyled = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin: 0.5em 0 0;
  font-weight: bold;
  font-size: 1.8rem;
`

const MovieGenreTag = styled.span`
  display: inline-flex;
  margin: 0.1em 0.4em 0.1em 0;
  color: #555555;
  font-size: 1.2rem;
  &:after {
    content: ', ';
  }
  &:last-child {
    &:after {
      display: none;
    }
  }
`

export default MovieGenreList
