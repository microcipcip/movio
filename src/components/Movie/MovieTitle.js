import React from 'react'
import PropTypes from 'prop-types' // eslint-disable-line no-unused-vars
import { MovieConsumer } from './Movie'

const MovieTitle = props => (
  <MovieConsumer>{({ title }) => <h2>{title}</h2>}</MovieConsumer>
)

MovieTitle.propTypes = {}
MovieTitle.defaultProps = {}

export default MovieTitle
