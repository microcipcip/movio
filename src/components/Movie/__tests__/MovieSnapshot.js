import React from 'react'
import { customRender, genreList, movieList } from 'utils/testUtils'
import Movie from '../Movie'

it('Should create snapshot', async () => {
  const movie = movieList[0]
  const { container } = customRender(
    <Movie genreList={genreList} {...movie}>
      <Movie.Cover />
      <Movie.Title />
      <Movie.Date />
    </Movie>,
    { genreList, movieList }
  )
  expect(container).toMatchSnapshot()
})
