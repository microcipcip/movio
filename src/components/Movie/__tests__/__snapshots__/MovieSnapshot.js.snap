// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should create snapshot 1`] = `
.c1 {
  position: relative;
}

.c1:after {
  content: '';
  display: block;
  padding-top: 150%;
}

.c2 {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  object-fit: cover;
  border-radius: 2px;
}

.c3 {
  margin: 0;
  font-weight: bold;
  font-size: 1.8rem;
}

.c4 {
  color: #c3c3c3;
  text-align: right;
  font-size: 1.2rem;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Venom"
        class="c2"
        src="https://image.tmdb.org/t/p/w342/2uNW4WbgBXL25BAbXGLnLqX71Sw.jpg"
      />
    </div>
    <h2
      class="c3"
    >
      Venom
    </h2>
    <time
      class="c4"
      datetime="2018-10-03"
    >
      03 10 2018
    </time>
  </div>
</div>
`;
